app/js/lib/foxbox/api.js

```diff
@@ -148,4 +222,78 @@ export default class API {
 
     return new Promise((resolve) => settings.once('session', () => resolve()));
   }
+
+  /**
+   * Fetches values for the set of getters.
+   *
+   * @return {Promise}
+   * @private
+   */
+  [p.fetchGetterValues]() {
+    // It may happen that all watchers have been unregistered in the meantime,
+    // so let's return early in this case.
+    if (this[p.watchGetters].size === 0) {
+      return Promise.resolve();
+    }
+
+    const selectors = Array.from(this[p.watchGetters].values()).map(
+      ({ id }) => ({ id })
+    );
+
+    return this.put('channels/get', selectors)
+      .then((response) => {
+        Object.keys(response).forEach((key) => {
+          const getter = this[p.watchGetters].get(key);
+          if (!getter) {
+            return;
+          }
+
+          this[p.updateGetterValue](getter, response[key]);
+        });
+      });
+  }
+
+  /**
+   * Updates getter value if needed. If value has changed, appropriate event is
+   * fired.
+   *
+   * @param {{ id: string, value: Object }} getter Getter to update value for.
+   * @param {Object} getterValue Getter value returned from the server.
+   *
+   * @private
+   */
+  [p.updateGetterValue](getter, getterValue) {
+    let valueChanged = false;
+
+    if (!getterValue || !getter.value) {
```

Comment:
That's a special case when state of the value is not known yet (happens for OpenZWave pretty frequently)